File: java/src/main/java/tech/qmates/openchat/domain/repository/UserRepository.java
package tech.qmates.openchat.domain.repository;

import tech.qmates.openchat.domain.entity.RegisteredUser;
import tech.qmates.openchat.domain.entity.UserToRegister;

import java.util.Set;
import java.util.UUID;

public interface UserRepository {
    void store(UserToRegister user);
    boolean isUsernameAlreadyUsed(String username);
    Set<RegisteredUser> getAll();
    RegisteredUser getUserById(UUID uuid);
    void reset();
}

File: typescript/src/domain/repositories/UserRepository.ts
import { RegisteredUser, UserToRegister } from "../entities/User"

export default interface UserRepository {
  store(user: UserToRegister): void
  isUsernameAlreadyUsed(username: string): boolean
  getAll(): RegisteredUser[]
  getUserById(userId: string): RegisteredUser | null
  reset(): void
}